source/includes/monitoring/monitoring.rb

```diff
@@ -0,0 +1,173 @@
+# start-available-subscriber
+subscriber = Mongo::Monitoring::ServerOpeningLogSubscriber.new
+
+# Globally subscribes to ServerOpening events by using the SERVER_OPENING monitoring topic
+Mongo::Monitoring::Global.subscribe(Mongo::Monitoring::SERVER_OPENING, subscriber)
+client = Mongo::Client.new(['127.0.0.1:27017'])
+
+# Subscribes to ServerOpening events at the client level by using the SERVER_OPENING monitoring topic
+client.subscribe( Mongo::Monitoring::SERVER_OPENING, subscriber )
+# end-available-subscriber
+
+# start-sdam
+class SDAMLogSubscriber
+    include Mongo::Loggable
+  
+    def succeeded(event)
+      log_debug(format_event(event))
+    end
+  
+    private
+  
+    def logger
+      Mongo::Logger.logger
+    end
+  
+    def format_message(message)
+      format("SDAM | %s".freeze, message)
```

Comment:
There's no need to explicitly freeze string literals. Most Ruby programs have the `# frozen_string_literal: true` comment at the top, which causes all string literals in the file to be automatically frozen by default. And even without that, the string is being used here in a way that makes it impossible for clients to mutate.

Generally, the intent of the code will be clearer if you omit the explicit `#freeze` call.